contractcourt/htlc_success_resolver.go

```diff
@@ -54,6 +54,12 @@ type htlcSuccessResolver struct {
 	// htlc contains information on the htlc that we are resolving on-chain.
 	htlc channeldb.HTLC
 
+	// isTapscriptRoot indicates whether the htlc is on a TapscriptRoot
+	// channel type.
+	// TODO: remove this when incoming HTLCs with custom records can be
+	// resolved for this channel type
+	isTapscriptRoot bool
```

Comment:
think we can already get the info using the scripts, copied from other PRs,
```go
// isTaproot returns true if the htlc output is a taproot output.
func (h *htlcTimeoutResolver) isTaproot() bool {
	return txscript.IsPayToTaproot(
		h.htlcResolution.SweepSignDesc.Output.PkScript,
	)
}
```

Reply:
IIUC `IsPayToTaproot` isn't a sufficient check here? Based on the original [comment](https://github.com/lightningnetwork/lnd/pull/8390#issuecomment-2420670676), I believe that we're specifically looking for `TapscriptRoot` channel types - this check would be true for both `TapcriptRoot` and `SimpleTaproot` channels because it's just checking that we have a `P2TR` witness?

Reply:
yeah you are right these are two different things. In that case do you think it's possible to pass the boolean when creating the resolvers so we don't need the `SupplementState` hack? I remembered we sort have to use this hack, just wanna double check (sth sth reloading from the disk I think).

Reply:
Yeah this is how we currently go about [relaunching](https://github.com/lightningnetwork/lnd/blob/master/contractcourt/channel_arbitrator.go#L740) after startup.  We don't have `state` with our `ChanType` on hand when we [read from disk](https://github.com/lightningnetwork/lnd/blob/master/contractcourt/briefcase.go#L575) which means we'll always need to supplementary step on restarts.

I think we should just always do it that way for consistency?

Reply:
yeah that's a design i'm trying to fix, pre-exiting so no worries.